File: backend/app/models/notificacion.py
from sqlalchemy import Column, Integer, ForeignKey, DateTime, String
from sqlalchemy.orm import relationship
from app.db.base_class import Base


class Notificacion(Base):
    __tablename__ = "notificacion"
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id")) 
    tipo = Column(String)                             
    mensaje = Column(String)                          
    fecha_envio = Column(DateTime)                    
                                                      
    user = relationship("User") 

File: backend/app/models/notificacion.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Notificacion extends Model {
    static associate(models) {
      Notificacion.belongsTo(models.User, {
        foreignKey: 'userId',
        as: 'user',
      });
    }
  }
  Notificacion.init({
    tipo: DataTypes.STRING,
    userId: DataTypes.INTEGER,
    mensaje: DataTypes.STRING,
    fecha_envio: DataTypes.DATEONLY,
    visto: DataTypes.BOOLEAN
  }, {
    sequelize,
    modelName: 'Notificacion',
  });
  return Notificacion;
};